packages/customize-widgets/src/components/error-boundary/index.js

```diff
@@ -26,6 +27,8 @@ export default class ErrorBoundary extends Component {
 
 	componentDidCatch( error ) {
 		this.setState( { error } );
+
+		doAction( 'gb.reportErrorBoundaryException', error );
```

Comment:
To avoid repeating the identifier string everytime, we might abstract this into a function and extract it in a module somewhere. Something like:

```js
export function extendErrorBoundaryDidCatch( error ) {
    doAction( 'gb.extendErrorBoundaryDidCatch', error );
}
```

You may see I also renamed it to `extendErrorBoundaryDidCatch`, the reason is that the action callback is not limited to only reporting the error to an external target source (though I assume this will be one of the main use cases), so semantically `extendErrorBoundaryDidCatch` makes more sense, I think.

Reply:
Another thing that we should discuss is whether this action will apply to all Error Boundaries or if we need more granularity. I'd say it should apply to all boundaries, at least to start with, and we can see if there's an actual need to provide more fine control.

Reply:
> It looks like all error boundaries do the same thing in all places and save the error in the state so we could as well extract a general React component that implements getDerivedStateFromError and componentDidCatch.

That's a good idea! I also briefly thought about that, but didn't want to deviate too much from the main goal. Would it be better to do that as a follow-up (I can track it and make sure we don't forget)?

Reply:
What do you think if we start with abstracting in a function, like I suggested [here](https://github.com/WordPress/gutenberg/pull/42024/files#r909107114)? This would also make it easier to document it in a single place. I'm just not sure what would be a good place for such a helper function, I'll poke around, but I'm open to suggestions! After that, we can then follow up with your suggestion about DRYing the boundaries.

Reply:
Would `@wordpress/compose` be a good package to host this function for now? We could name it `compose.ErrorBoundary.error`, as you suggested.

Reply:
I did some code checking. `@wordpress/compose` doesn't depend on `@wordpress/hooks` package and I think we should keep it general enough. I see that `@wordpress/block-editor` uses WP filters already so it's probably better to put this functionality there. We can take care of it later though. For now, we only need to settle on the name of the hook.